internal/commands/pull/command.go

```diff
@@ -168,30 +197,40 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 	}
 	ui.Print(terminal.NewTextLog("Saved app to disk"))
 
-	if cmd.inputs.IncludeDependencies {
+	if cmd.inputs.IncludeNodeModules || cmd.inputs.IncludePackageJSON || cmd.inputs.IncludeDependencies {
+		logStr := "as a node_modules archive"
+		if cmd.inputs.IncludePackageJSON {
+			logStr = "as a package.json file"
+		}
+
 		s := spinner.New(terminal.SpinnerCircles, 250*time.Millisecond)
-		s.Suffix = " Fetching dependencies archive..."
+		s.Suffix = fmt.Sprintf(" Fetching dependencies %s...", logStr)
 
 		exportDependencies := func() error {
 			s.Start()
 			defer s.Stop()
 
-			archiveName, archivePkg, err := clients.Realm.ExportDependencies(app.GroupID, app.ID)
+			fileName, file, err := cmd.exportDependencies(clients, app)
 			if err != nil {
 				return err
 			}
 
+			if cmd.inputs.IncludePackageJSON && fileName != local.NamePackageJSON {
+				logStr = "as a node_modules archive"
+				ui.Print(terminal.NewWarningLog("The package.json file was not found, exporting a node_modules archive instead"))
```

Comment:
also, can you test this out locally to see how it displays?

something tells me that if you `ui.Print` while we still have `s.Start()` running, that we'll end up with some wonky output where the spinner will end up showing twice on screen (once at the bottom of the buffer, and once as added to the line of the warning log you just printed)

we may need to have something like:

```go
var packageJSONMissing bool

exportDependencies := func() error {
  s.Start()
  defer s.Stop()

  // existing code

  if cmd.inputs.IncludePackageJSON && fileName != local.NamePackageJSON {
    packageJSONMissing = true
  }

  return local.WriteFile(...)
}

if packageJSONMissing {
  logStr = "as a node_modules archive"
  ui.Print(terminal.NewWarningLog("The package.json file was not found, exporting a node_modules archive instead"))
}
```

if that _is_ the case, could you add a bullet point to the "Long-Running Processes" section of `guidelines.md` as well?

Reply:
![Screen Shot 2021-09-13 at 4 40 43 PM](https://user-images.githubusercontent.com/20569753/133160187-c89cff54-11cb-4284-a051-fee16ce7d357.png)
Did a quick check, it looks like this.

Reply:
yeah so we want to make sure that warning message prints on its own line...

can we go with my suggested approach above, to defer printing that warning log until after we're fully done with the spinner?

Reply:
I made the change but since it did not show up twice. So I think no need to update the `guidelines.md` file as all behaviors here seems expected.

Reply:
we should update the `guidelines.md` to call this out, so hopefully future devs won't run into the same thing you did here:

can you add this as the 3rd bullet point in that "Long Running Processes" section:

```cmd
* avoid printing messages (with the "UI") _during_ the display of the spinner as it will lead to messages not appearing on their own line, instead defer the printing of any informational/warning messages until after the spinner has been stopped and removed
```